tests/simulator_test.cpp

```diff
@@ -132,25 +122,28 @@ TEST_F(EnergyCollectorTest, NotEvenNumOfEnergyCollectorTest) {
 
   Ray alongX(kVecZero, kVecX);
   ASSERT_TRUE(performHitCollector(energyCollectors, alongX, &hitData));
-  ASSERT_EQ(collectorPositionRadius - refCollectorRadius, hitData.time);
+  ASSERT_FLOAT_EQ(collectorPositionRadius - refCollectorRadius, hitData.time);
 
   Ray alongY(kVecZero, kVecY);
   ASSERT_TRUE(performHitCollector(energyCollectors, alongY, &hitData));
-  ASSERT_EQ(collectorPositionRadius - refCollectorRadius, hitData.time);
+  ASSERT_FLOAT_EQ(collectorPositionRadius - refCollectorRadius, hitData.time);
 
-  Ray at2Angle(kVecZero, Vec3(std::cos(2 * collectorAngle), 0,
-                              std::sin(2 * collectorAngle)));
+  const float collectorAngle = 2 * kPi / (numCollectors - 1);
+  Ray at2Angle(kVecZero, Vec3(std::cos(0.5 * collectorAngle), 0,
+                              std::sin(0.5 * collectorAngle)));
   ASSERT_TRUE(performHitCollector(energyCollectors, at2Angle, &hitData));
   // TODO: This case doesn't work, find out why
-  // ASSERT_EQ(collectorPositionRadius - refCollectorRadius, hitData.time);
+  // printCollectors(energyCollectors);
+  // ASSERT_FLOAT_EQ(collectorPositionRadius - refCollectorRadius, hitData.time);
```

Comment:
Why do you expect the collision happen exactly at radius distance from the center of a collector?

It looks to me like you are aiming at the middle point between 2 collectors, so collision time will be larger than this (am I missing something?)

You said you visualized it? How?

Reply:
Yes, or maybe you got test code wrong. Not sure yet.